seeq/addons/correlation/utils/_sdl.py

```diff
@@ -71,3 +72,24 @@ def get_workbook_worksheet_workstep_ids(url):
     if 'workstepId' in params:
         workstep_id = params['workstepId'][0]
     return workbook_id, worksheet_id, workstep_id
+
+
+def is_ipv4(string):
+    parsed_url = parse_url(string)
+    try:
+        ipaddress.IPv4Network(parsed_url.hostname)
+        return True
+    except ValueError:
+        return False
+
+
+def get_seeq_url():
+    if hasattr(spy.session, 'public_url'):
+        if not is_ipv4(spy.session.public_url):
```

Comment:
I recently noted for a different project that there is a chance that `spy.session.public_url` or `spy.session.private_url` are None. If that's the case, this line will fail. I suggest adding a check to both the public and private code paths. If they are None just skip, if both are None the function will return None